heap.py

```diff
@@ -0,0 +1,36 @@
+class heap:
+
+    def __init__(self, array):
+        self.array = array
+        self.position = (len(array)-1) #tracks the position of the part of the list that isn't solved
+        
+    def heapify(self):
+        #left son - 2n
+        #right son - 2n+1
+        def swaps():
+            for n in range(int((self.position+1)/2)):
+                if self.array[n] < self.array[2*n]:
+                    #swap parent with left son
+                    self.array[n], self.array[2*n] = self.array[2*n], self.array[n]  
+                    swaps()
+                elif self.array[n] < self.array[2*n+1]:
+                    #swap parent with right son
+                    self.array[n], self.array[2*n+1] = self.array[2*n+1], self.array[n]
+                    swaps()
+        swaps()
+        
+    def max_heap(self):
```

Comment:
This function builds the heap and do many complex stuff.
Instead, keep the heap property from the minute you create it, then pop_max will be just poping the first index.